inc/packages/admin/namespace.php

```diff
@@ -0,0 +1,189 @@
+		[],
+		FAIR\VERSION
+	);
+}
+
+function render_tab_direct() {
+	?>
+	<div class="fair-direct-install">
+		<p class="fair-direct-install__help">
+			<?= __( "If you have a plugin's ID, enter it here to view the details<br />and install it.", 'fair' ); ?>
+		</p>
+		<form
+			action=""
+			class="fair-direct-install__form"
+			method="post"
+		>
+			<input
+				type="hidden"
+				name="tab"
+				value="<?= esc_attr( TAB_DIRECT ) ?>"
+			/>
+			<?php wp_nonce_field( TAB_DIRECT ); ?>
+			<label
+				class="screen-reader-text"
+				for="plugin_id"
+			>Plugin ID</label>
+			<input
+				type="text"
+				id="plugin_id"
+				name="plugin_id"
+				pattern="did:(web|plc):.+"
+				placeholder="did:..."
+			/>
+			<?php submit_button( _x( 'View Details', 'fair' ), '', '', false ); ?>
+		</form>
+		<p class="fair-direct-install__note">
+			<?= __( 'Plugin IDs should be in the format <code>did:web:...</code> or <code>did:plc:...</code>', 'fair' ); ?>
+		</p>
+	</div>
+	<script>
+		// On submit, trigger the thickbox with the plugin information.
+		document.querySelector( '.fair-direct-install__form' ).addEventListener( 'submit', function ( e ) {
+			e.preventDefault();
+			e.stopPropagation();
+			const id = document.getElementById( 'plugin_id' ).value;
+
+			// Get the current URL without the query string.
+			const currentUrl = new URL( window.location.href );
+
+			// Construct the URL for the plugin information page.
+			const baseUrl = currentUrl.origin + currentUrl.pathname;
+			const params = new URLSearchParams( {
+				tab: 'plugin-information',
+				plugin: id,
+				TB_iframe: 'true',
+				width: '600',
+				height: '550'
+			} );
+
+			const url = baseUrl + '?' + params.toString();
+			tb_show( null, url, false );
+
+			// Set ARIA role, ARIA label, and add a CSS class.
+			const tbWindow = jQuery( '#TB_window' );
+			tbWindow
+				.attr({
+					'role': 'dialog',
+					'aria-label': wp.i18n.__( 'Plugin details' )
+				})
+				.addClass( 'plugin-details-modal' );
+
+			// Set title attribute on the iframe.
+			tbWindow.find( '#TB_iframeContent' ).attr( 'title', wp.i18n.__( 'Plugin details' ) );
+		} );
+	</script>
+	<?php
+}
+
+function get_direct_install_url( MetadataDocument $doc, ReleaseDocument $release ) {
+	$args = [
+		'action' => ACTION_INSTALL,
+		'id' => urlencode( $doc->id ),
+		'version' => urlencode( $release->version ),
+	];
+	$url = add_query_arg( $args, self_admin_url( 'update.php' ) );
+	return wp_nonce_url( $url, ACTION_INSTALL_NONCE . $doc->id );
+}
+
+function handle_direct_install() {
+	$id = wp_unslash( $_GET['id'] );
+	check_admin_referer( ACTION_INSTALL_NONCE . $id );
+
+	$version = wp_unslash( $_GET['version'] ?? null );
+	if ( empty( $version ) ) {
+		wp_die( __( 'No version specified for the plugin.', 'fair' ) );
+	}
+
+	$skin = new \WP_Upgrader_Skin();
+	$res = Packages\install_plugin( $id, $version, $skin );
+	var_dump( $res );
+	exit;
+}
+
+function embedded_info_page() {
```

Comment:
Is this function used anywhere?

Reply:
No; the idea originally was for this to display the info page, but I discovered the behaviour was tied heavily into the iframe/modal, so skipped it.